src/peft/tuners/ia3/model.py

```diff
@@ -299,3 +300,104 @@ def merge_and_unload(self):
                 setattr(parent, target_name, target.modules_to_save[target.active_adapter])
 
         return self.model
+
+    def delete_adapter(self, adapter_name: str):
+        """
+        Deletes an existing adapter.
+
+        Args:
+            adapter_name (str): Name of the adapter to be deleted.
+        """
+        if adapter_name not in list(self.peft_config.keys()):
```

Comment:
```suggestion
        if adapter_name not in self.peft_config:
```

I know this is just a 1:1 copy from LoRA, but let's do it better here and avoid the costly list.